webpack/assets/javascripts/react_app/components/statistics/ChartBox.js

```diff
@@ -29,30 +31,29 @@ class ChartBox extends React.Component {
   }
 
   render() {
-    const { chart } = this.props;
-
-    const modalConfig = getLargePieChartConfig({
-      data: this.props.chart.data,
-      id: `${chart.id}Modal`,
-    });
-
+    const { chart, type } = this.props;
+    const components = {
+      donut: DonutChart,
+    };
+    const Chart = components[type];
```

Comment:
Why using the `components` map instead of just use the `DonutChart`?

Reply:
IMO ChartBox should be generic, and be able to support different kind of charts, for later use.

Reply:
So can you set the `propType.type` to be `oneOf(['donut'])`?